torch/onnx/_internal/exporter/_torchlib/ops/nn.py

```diff
@@ -0,0 +1,71 @@
+"""torch.ops.aten operators under the `core` module."""
+# mypy: disable-error-code="misc,arg-type,type-arg,valid-type,assignment,return-value,type-var,operator,no-untyped-def,index"
+# ruff: noqa: TCH001,TCH002
+# flake8: noqa
+
+from __future__ import annotations
+
+import math
+
+from onnxscript.onnx_opset import opset18 as op, opset20 as op20
+
+import torch
+from torch.onnx._internal.exporter._torchlib._tensor_typing import TReal
+from torch.onnx._internal.exporter._torchlib._torchlib_registry import onnx_impl
+
+
+_MATH_PI = math.pi
+
+
+aten = torch.ops.aten
+
+
+@onnx_impl(aten.gelu.default, trace_only=True)
+def aten_gelu(self: TReal, approximate: str = "none") -> TReal:
+    """gelu(Tensor self, *, str approximate='none') -> Tensor"""
+
+    if approximate == "tanh":
+        result = _aten_gelu_approximate_tanh(self)
+    else:
```

Comment:
should we check if approximate == "none"? And raise error otherwise?

Reply:
Else, we will export an invalid onnx model with invalid attribute value

Reply:
seems like torch.gelu protects against this case, and doesn't allow any other str than none or tanh
https://github.com/pytorch/pytorch/blob/2e23768d2521acec9314d07a4b418089848a0f26/torch/_refs/nn/functional/__init__.py#L1079C9-L1079C80

Reply:
Ok. Maybe Justin can clarify what the convention is: do we assume that the inputs to these functions satisfy torch implementation requirements or do we check anyway?

Reply:
Yes we can assume valid inputs because that is what we get from the fx graph. The fx graph (from exported program) is always valid.